app/core/modals/PublishModuleModal.tsx

```diff
@@ -90,33 +91,171 @@ export default function PublishModule({ module, user, workspace }) {
                   <>
                     <div className="mt-2">
                       <p className="text-base text-gray-500 dark:text-gray-300">
-                        Once you publish this module, you cannot delete it. You can publish this
-                        module for free.
+                        Once you publish this module, you cannot delete it.
                       </p>
+                      <p className="text-base text-gray-500 dark:text-gray-300">
+                        You can publish this module for free or support us with an amount of your
+                        choice (incl. VAT).
+                      </p>
+                      <span className="my-2 flex w-full flex-col">
+                        <input
+                          id="pay-what-you-want"
+                          type="range"
+                          className="pwyw"
+                          value={payWhat}
+                          max={100}
+                          onChange={(value) => {
+                            setPayWhat(parseInt(value.target.value))
+                          }}
+                        ></input>
+                        <label className="sr-only" htmlFor="pay-what-you-want">
+                          Pay what you want price
+                        </label>
+                        <span className="my-2 flex text-sm text-gray-500 dark:text-gray-200">
+                          I want to support Researchequals with
+                          <p className="flex">
+                            <span className="inline-block h-full align-middle"></span>
+                            <p className="my-auto mx-1 inline-block align-middle text-base">
+                              &euro;
+                            </p>
+                            <input
+                              type="text"
+                              id="name"
+                              name="name"
+                              required
+                              value={payWhat}
+                              className="placeholder-font-normal block w-full appearance-none rounded-md border border-gray-300 px-3 py-2 text-sm font-normal placeholder-gray-400 shadow-sm focus:border-indigo-500 focus:outline-none focus:ring-indigo-500  dark:border-gray-500 dark:bg-gray-800"
+                              onChange={(value) => {
+                                setPayWhat(parseInt(value.target.value) || 0)
```

Comment:
I think (if your goal is to get more people to pay you (which I think you should!)) it might be a good idea to set the default at a non-zero value, like idk 5/10 euros or something. This does sorta induce some kind of implicit expectation that you should pay, kind of like how bandcamp (used to?) do that, but I don't think that'll deter people or anything.

Reply:
Thanks @tefkah for taking the time and making this suggestion!

I think that's a great idea. Will reach out to our testers and consider how we can test this empirically moving forward!

Reply:
Came out with a 7-5 preference for non-zero default - zero default, respectively (based on tester feedback).

Will update the pull request to default at 5EUR 😊

Reply:
Awesome!